noir-projects/noir-protocol-circuits/crates/types/src/merkle_tree/variable_merkle_tree.nr

```diff
@@ -72,40 +71,42 @@ impl VariableMerkleTree {
     // |       tx_0      |  | tx_1 |
     //
     pub fn new_sha<let N: u32>(leaves: [Field; N], num_non_empty_leaves: u32) -> Self {
-        let prev_power_2 = get_prev_power_2(num_non_empty_leaves);
-
-        // hash base layer
-        // If we have no num_non_empty_leaves, we return 0
-        let mut stop = num_non_empty_leaves == 0;
-
+        let num_nodes_layer_1 = if (num_non_empty_leaves == 0) {
```

Comment:
Renamed `prev_power_2` to `num_nodes_layer_1` as I found it hard to reason about `prev_power_2`.